### PR TITLE
Consistent log messages

### PR DESCRIPTION
**Pull request by adams85**
### Describe the purpose of your pull request

Revises log messages to make them consistent with other SDKs and also adds event ids for easier log event identification.

Note: I haven't implemented structured logging but the [logging abstraction supports it](https://pkg.go.dev/github.com/sirupsen/logrus@v1.9.0). However, it would require a BC to our `Logger` interface AFAICS.

### Requirement checklist (only if applicable)

- [ ] I have covered the applied changes with automated tests.
- [ ] I have executed the full automated test set against my changes.
- [ ] I have validated my changes against all supported platform versions.
- [ ] I have read and accepted the [contribution agreement](https://github.com/configcat/legal/blob/main/contribution-agreement.md).

### CODE REVIEW

File: v7/snapshot.go [hunk 1]
```diff
@@ -209,15 +209,15 @@ func (snap *Snapshot) details(id keyID, key string) (interface{}, string, *wirec
 		eval = snap.evaluators[id]
 	}
 	if eval == nil {
-		err := fmt.Errorf("error getting value: value not found for key %s."+
-			" Here are the available keys: %s", key, strings.Join(snap.GetAllKeys(), ","))
-		snap.logger.Errorf("%v", err)
-		return nil, "", nil, nil, err
+		var message = "Failed to evaluate setting '%s' (the key was not found in config JSON). Available keys: %s."
+		var availableKeys = strings.Join(snap.GetAllKeys(), ", ")
```
Comment (adams85) [inline v7/snapshot.go:213]: Available keys should be surrounded with single quotes like `'Key1', 'Key2', 'Key3'`

File: v7/config_fetcher.go [hunk 2]
```diff
@@ -129,7 +129,7 @@ func (f *configFetcher) runPoller(pollInterval time.Duration) {
 			return
 		}
 		if err := f.refreshIfOlder(f.ctx, time.Now().Add(-pollInterval/2), true); err != nil {
-			f.logger.Errorf("cannot refresh configcat configuration: %v", err)
+			f.logger.Errorf(0, "cannot refresh configcat configuration: %v", err)
```
Comment (adams85) [inline v7/config_fetcher.go:132]: This SDK uses a bunch of log messages which are not present in other SDKs. I haven't introduce event ids for these. (Which is probably OK.) However, these messages is mostly lowercased and lacks dot at the end, which we should probably make consistent with the rest of the messsages.

File: v7/config_fetcher.go [hunk 3]
```diff
@@ -359,7 +382,11 @@ func (f *configFetcher) fetchHTTPWithoutRedirect(ctx context.Context, baseURL st
 	request = request.WithContext(f.ctx)
 	response, err := f.client.Do(request)
 	if err != nil {
-		return nil, fmt.Errorf("config fetch request failed: %v", err)
+		var message = "Unexpected error occurred while trying to fetch config JSON: %v"
```
Comment (adams85) [inline v7/config_fetcher.go:385]: Request timeout (1102, `Request timed out while trying to fetch config JSON. Timeout value: {TIMEOUT}`) is not handled here.  I'm not sure it's an oversight or is just not possible in this case.

**Review by adams85 — COMMENTED** [review]
Logging of error #1000 (ConfigJsonIsNotPresent) is missing from this SDK and error #1001 (SettingEvaluationFailedDueToMissingKey) might be incomplete as well. I'm not sure how to set this straight because Go is pretty unfamiliar to me and this implementation differs from other SDKs to a great extent. So I'll leave this up to the author of the SDK.

File: v7/snapshot.go (hunk 1 above)
Comment (z4kn4fein) [reply v7/snapshot.go]: Done

File: v7/config_fetcher.go (hunk 3 above)
Comment (z4kn4fein) [reply v7/config_fetcher.go]: Done

**Review by z4kn4fein — APPROVED** [review]

**Pull request merged**